packages/utils/src/schema/getMatchingOption.ts

```diff
@@ -32,6 +33,12 @@ export default function getMatchingOption<
   if (formData === undefined) {
     return 0;
   }
+
+  const manualDiscriminatorMatch = getOptionMatchingManualDiscriminator(discriminatorField, formData, options);
+  if (manualDiscriminatorMatch !== undefined) {
+    return manualDiscriminatorMatch;
+  }
+
```

Comment:
This implementation already exists on lines `46-51`. Are you trying to avoid the `isValid` check?

Reply:
Yes, this is exactly what I am doing - this makes all the difference between a fast vs. an unusably sluggish form in my use-case :)

Reply:
I'm wondering if the [PR fix](https://github.com/rjsf-team/react-jsonschema-form/pull/3721) will help your use case once we release it?